sdk/script/management-sdk.ts

```diff
@@ -8,6 +8,10 @@ import Promise = Q.Promise;
 
 import { AccessKey, Account, App, CodePushError, CollaboratorMap, CollaboratorProperties, Deployment, DeploymentMetrics, Headers, Package, PackageInfo, UpdateMetrics } from "./types";
 
+var superproxy = require("superagent-proxy");
+superproxy(superagent);
+var proxy = process.env.HTTPS_PROXY || process.env.https_proxy || process.env.HTTP_PROXY || process.env.http_proxy;
```

Comment:
I prefer not to have the Management SDK read environment variables directly, because it is designed to be used as a general library. I prefer for the proxy behaviour be explicitly determined by the application using it.

Would this behaviour be more transparent if the `proxy` setting is passed to in the constructor? We can have the CLI read the environment variable and pass the proxy setting to it instead. Either that, or have a `code-push login --proxy <proxyUrl>` option, which saves the proxy setting in the login credentials. What do you think, @silhouettes